pkg/editors/lookup.go

```diff
@@ -0,0 +1,70 @@
+package editors
+
+import (
+	"context"
+	goerrors "errors"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/cockroachdb/errors"
+)
+
+// FoundEditor is a found external editor on the user's machine
+type FoundEditor struct {
+	// The friendly name of the editor, to be used in labels
+	Editor string `json:"editor"`
+	// The executable associated with the editor to launch
+	Path string `json:"path"`
+	// The editor requires a shell to spawn
+	UsesShell bool `json:"usesShell,omitempty"`
+}
+
+var (
+	editorCacheMu sync.Mutex
+	// editorCache is a cache of the available editors on the user's machine
+	editorCache []FoundEditor
+
+	// ErrEditorNotFound is returned when an editor is not found when called from
+	// the Find function
+	ErrEditorNotFound = goerrors.New("editor not found")
+)
+
+// Resolve a list of installed editors on the user's machine, using the known
+// install identifiers that each OS supports.
+func Resolve(ctx context.Context) ([]FoundEditor, error) {
+	editorCacheMu.Lock()
+	defer editorCacheMu.Unlock()
+
+	// If we don't have a cache yet, populate it
+	if len(editorCache) == 0 && cap(editorCache) == 0 {
```

Comment:
This would be nicer with https://pkg.go.dev/go4.org/syncutil#Once

Reply:
Hmm, this is already being performed under a lock, `syncutil.Once` would just introduce a new lock and new variable. Is this a readability thing for you?

Reply:
I think with `syncutil.Once` we wouldn't need a separate lock, but up to you. Yes, just for readability